src/IronRure/Options.cs

```diff
@@ -1,38 +1,61 @@
 using System;
 
-namespace IronRure
+namespace IronRure;
+
+/// <summary>
+/// Regex compilation options.
+/// </summary>
+public sealed class Options : IDisposable
 {
-    /// <summary>Regex compilation options.</summary>
-    public class Options : IDisposable
+    private bool _disposed;
+
+    /// <summary>
+    /// Creates a new options instance with the default values.
+    /// </summary>
+    public Options()
     {
-        /// <summary>Create a new options instnace with the default values.</summary>
-        public Options()
-        {
-            Raw = RureFfi.rure_options_new();
-        }
+        Raw = RureFfi.rure_options_new();
+    }
 
-        internal OptionsHandle Raw { get; }
+    internal OptionsHandle Raw { get; }
 
-        /// <summary>
-        ///   Set Size Limit - Controls the size of a single regex program
-        /// </summary>
-        public uint Size
+    /// <summary>
+    /// Sets the size limit for a single regex program.
+    /// </summary>
+    public uint Size
+    {
+        set
         {
-            set => RureFfi.rure_options_size_limit(Raw, new UIntPtr(value));
+            if (_disposed)
```

Comment:
I think the throw if is preferred. Keeps the throw out of the current method which reduces code size of the method, means throws are less likely to be branch predicted, and leads to better instruction cache usage.

Reply:
I think I've sorted it to what you meant by changing it to  ObjectDisposedException.ThrowIf(_disposed, nameof(Options));